### DISCUSSION TITLE
SunTouch only responds to CIRCUIT_SET messages, and only if I change the src... anybody else deal with this?

### DISCUSSION THREAD

**Discussion by jaroy** [cvonk/OPNpool#5] [Q&A] [answered]
I have been testing with my SunTouch, but I had to set `src` to 0x48 in the header to get the controller to ack `CIRCUIT_SET` messages (same as https://docs.google.com/document/d/1M0KMfXfvbszKeqzu6MUF_7yM6KDHk8cZ5nrH1_OUcAc/edit).

However, it doesn't respond to any other `_SET` or `_REQUEST` messages, no matter what value I set `src` to.

Anybody else had this issue? Does something have to be enabled (menus, dip switches, etc) to make the controller believe there is a connected remote perhaps? Couldn't find anything in the Pentair user manual, but that's not saying much.

Firmware is 2.050, which is a few versions behind the one mentioned in the README. How do you update SunTouch firmware 🤣

**Comment by jaroy** [comment, marked as answer]
Updating the firmware resolved this.

> Firmware is 2.050... How do you update SunTouch firmware 🤣

The answer to this was surprisingly easy https://www.troublefreepool.com/threads/how-to-perform-firmware-upgrade-on-pentair-controllers.69589/post-2318162. Only hard part was waiting 5 weeks for the USBDM programmer to arrive from China.

**Comment by cvonk** [comment]
Good to know you can upgrade the firmware. When we bought the house the
previous owner has just blown up the controller (by connecting it to 220V
instead of 110V), so I ended up with newish firmware.

Would be interesting to know when Pentair introduced what firmware. My
guess is that it was to support new devices such as remote control.

/C
